src/backend/distributed/executor/multi_utility.c

```diff
@@ -181,6 +181,23 @@ multi_ProcessUtility(Node *parsetree,
 		return;
 	}
 
+	if (IsA(parsetree, ExplainStmt))
+	{
+		ExplainStmt *explainStmt = (ExplainStmt *) parsetree;
+
+		if (IsA(explainStmt->query, Query))
+		{
+			Query *query = (Query *) explainStmt->query;
+
+			if (query->commandType == CMD_UTILITY &&
```

Comment:
Just asking blindly: Can't we call `ExtractFirstDistributedTableId(query)` and check for distributed tables?